spec/controllers/users/registrations_controller_spec.rb

```diff
@@ -1,22 +1,221 @@
-RSpec.describe Users::RegistrationsController do
+require 'rails_helper'
+
+RSpec.describe Users::RegistrationsController, type: :controller do
+  set_devise_mapping_to_user
+  render_views
+
   describe 'GET #index' do
-    it 'works' do
-      pending 'TODO'
-      fail
+    let(:do_request) { get :index }
+
+    shared_examples 'it displays volunteer' do
+      it 'displays volunteer' do
+        expect(response.body).to include(user.name)
+      end
+    end
+
+    shared_examples 'it does not display volunteer' do
+      it 'does not display volunteer' do
+        expect(response.body).to_not include(user.name)
+      end
+    end
+    
+    context 'with one volunteer' do
+      let!(:user) { create(:user_visible) }
+      
+      context 'with no sign-in' do
+        before { do_request }
+
+        it 'shows/assigns search bar' do
+          expect(assigns(:show_search_bar)).to eq(true)
```

Comment:
I'm stuck with using `assigns` here (and in a few of the following tests) as I couldn't find a way to test that the search bar/filter/etc features appear in the `response.body`